processing/src/main/java/org/apache/druid/segment/DimensionHandlerUtils.java

```diff
@@ -381,8 +386,12 @@ public static Comparable<?> convertObjectToType(
         switch (type.getElementType().getType()) {
           case STRING:
             return convertToComparableStringArray(obj);
-          default:
-            return convertToList(obj);
+          case LONG:
+            return convertToListWithObjectFunction(obj, CONVERT_OBJECT_TO_LONG_FUNCTION);
```

Comment:
nit: any reason not to use `DimensionHandlerUtils::convertObjectToLong` instead of static fields?

Reply:
This seems better